pages/category.page.js

```diff
@@ -0,0 +1,61 @@
+let WebPage = require("../base/webPage");
+
+let { WebButton } = require("../elemnets/button");
+let { WebView } = require("../elemnets/view");
+
+let baseElementLocator = by.css('.content_scene_cat');
+let listButtonLocator = by.css('#list');
+let addToCartButtonLocator = by.css('.ajax_add_to_cart_button');
+let cartModalWindowLocator = by.css('#layer_cart[style*="display: block"]');
+let checkoutButtonLocator = by.css('[title="Proceed to checkout"]');
+
+
+class CategoryPage extends WebPage {
+    constructor() {
+        super();
+    }
+
+    async clickListView() {
+        await allure.createStep('Click on "List" view button', async () => {
+            await this.getListElement().click();
+        })();
+    }
+
+    async clickAddToCart() {
+        await allure.createStep('Click on "Add to cart" button', async () => {
+            await this.getAddToCartElement().click();
+        })();
+    }
+
+    async cartModalWindowIsOpened() {
+        await browser.wait(protractor.ExpectedConditions.visibilityOf(this.getCartModalWindowElement().protractorElement), 20000);
```

Comment:
too complicated, you can just use `this.getCartModalWindowElement().waitForVisible (20000)`